core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -54,13 +61,17 @@
 public abstract class AbstractClient implements Client {
   private static final Logger LOG = LoggerFactory.getLogger(AbstractClient.class);
 
+  // RetryPolicy Supplier for business rpc calls
   private final Supplier<RetryPolicy> mRetryPolicySupplier;
 
   protected InetSocketAddress mAddress;
```

Comment:
Can we make this `Optional<InetSocketAddress>`?

Reply:
On closer look, why is this nullable?

Reply:
`DefaultBlockWorkerClient` doesn't have an `InetSocketAddress`. All it has is a `GrpcServerAddress`, which might contain a domain socket.

For other client like `AbstractMasterClient`, `mAddress` is not initialized until it queries the `MasterInquireClient`, so it could also be `null` at some point in the client's life time.